Tests/RevenueCatUITests/PaywallFooterTests.swift

```diff
@@ -151,11 +151,76 @@ class PaywallFooterTests: TestCase {
         expect(error).toEventually(matchError(Self.failureError))
     }
 
+    func testExternalRestoreHandler() throws {
+        var completed = false
+        var restoreCodeExecuted = false
+
+        let handler = Self.externalPurchaseHandler(performPurchase: { _ in
+            return (userCancelled: true, error: nil)
+        }, performRestore: {
+            restoreCodeExecuted = true
+            return (success: true, error: nil)
+        })
+
+        try Text("")
+            .paywallFooter(
+                offering: Self.offering,
+                customerInfo: TestData.customerInfo,
+                introEligibility: .producing(eligibility: .eligible),
+                purchaseHandler: handler
+            )
+            .addToHierarchy()
+
+        Task {
+            _ = try? await handler.restorePurchases()
+            completed = true
+        }
+
+        expect(completed).toEventually(beTrue())
+        expect(restoreCodeExecuted).to(beTrue())
+    }
+
+    func testExternalPurchaseHandler() throws {
+        var completed = false
+        var purchaseCodeExecuted = false
+
+        let handler = Self.externalPurchaseHandler(performPurchase: { _ in
+            purchaseCodeExecuted = true
+            return (userCancelled: true, error: nil)
+        }, performRestore: {
+            return (success: true, error: nil)
+        })
+
+        try Text("")
+            .paywallFooter(
+                offering: Self.offering,
+                customerInfo: TestData.customerInfo,
+                introEligibility: .producing(eligibility: .eligible),
+                purchaseHandler: handler
+            )
+            .addToHierarchy()
+
+        Task {
+            _ = try? await handler.purchase(package: TestData.packageWithIntroOffer)
+            completed = true
+        }
+
+        expect(completed).toEventually(beTrue())
```

Comment:
if we make the test method async, we can use await here, right? 
asking because `toEventually` does polling which can be slower, especially when the expectation is not met, so avoiding it might save some test execution time

Reply:
same for all other instances of toEventually

Reply:
These work and I have made the change. They work because we aren't relying on preference key changes, so the order is guaranteed.